docs/testing.md

```diff
@@ -0,0 +1,212 @@
+# Automated Testing in the MetaMask Extension Repository
+
+The purpose of this document is to summarize the testing tactics we want to align on for this repository.
+
+In its current version, this document does not include discussions on manual testing, performance testing, or browser testing. Our thoughts on our e2e tests will be shared on a separate file (to be linked here).
+
+To contribute, submit a PR. We encourage you to read the section about fitness functions below and include automation in your contribution.
+
+## Why We Test
+
+Testing is important because it helps ensure that software works as intended.
+
+Without automated tests, bugs organically occur in production or during manual regression testing for release candidates.
+
+This can lead to user dissatisfaction and costly rework. It also carries a big process and organizational overhead because people have to stop what they are doing multiple times and reach out to each other to uphold correctness.
+
+Speeding up the feedback loop from writing software to discovering bugs directly increases the productivity of the software delivery team. Apart from that, it minimizes context switching, increasing developer happiness and, indirectly, developer productivity.
+
+Testing is also proven to teach developers to write better and more readable code. Unit tests, for example, encourage separating functionality into smaller units that do less.
+
+Tests are generally the best way to document code. They are better than comments because they don't grow stale. If a block of code is moved or changed, developers must remember to update the corresponding comments. However, automated tests will break, and it'll be evident and impossible to miss what needs fixing by the developer.
+
+## Best Practices & Recommendations
+
+### Favor Jest instead of Mocha
+
+For consistency in syntax and coverage tooling, we favor using Jest for all unit tests. That means we don't need to use Sinon for mocks and stubs or Assert for assertions and that we'll eventually not need Mocha to run the tests. Jest replaces functionality from all these dependencies in one package.
+
+While Mocha is a test runner that can be used for unit, integration, and end-to-end testing, jest has been built specifically for unit testing and is much faster.
+
+To get a sense of the syntactic differences between the two approaches, you can see this [example PR](https://github.com/MetaMask/metamask-extension/pull/17226/files).
+
+### Tests Should Mirror How Software is Used
```

Comment:
I really love this advice, and that you've linked the testing library principles :clap: